cmd/autoconfigbrancher/main.go

```diff
@@ -155,66 +144,85 @@ func main() {
+				for _, fr := range o.futureReleases.Strings() {
+					args = append(args, []string{"--future-release", fr}...)
+				}
+				args = append(args, "--confirm")
+				return args
+			}(),
+		},
+		{
+			command:   "/usr/bin/ci-operator-prowgen",
+			arguments: []string{"--from-dir", "./ci-operator/config", "--to-dir", "./ci-operator/jobs"},
+		},
+
+		{
+			command:   "/usr/bin/ci-operator-config-mirror",
+			arguments: []string{"--config-path", "./ci-operator/config", "--to-org", "openshift-priv"},
+		},
+
+		{
+			command:   "/usr/bin/ci-operator-prowgen",
+			arguments: []string{"--from-dir", "./ci-operator/config", "--to-dir", "./ci-operator/jobs"},
+		},
+		{
+			command:   "/usr/bin/private-prow-configs-mirror",
+			arguments: []string{"--release-repo-path", "."},
+		},
+		{
+			command:   "./hack/ensure_job_cluster.py",
+			arguments: []string{"-w=true"},
+		},
+		{
+			command:   "/usr/bin/sanitize-prow-jobs",
+			arguments: []string{"--prow-jobs-dir", "./ci-operator/jobs"},
+		},
+	}
+
+	stdout := bumper.HideSecretsWriter{Delegate: os.Stdout, Censor: sa}
+	stderr := bumper.HideSecretsWriter{Delegate: os.Stderr, Censor: sa}
 	author := fmt.Sprintf("%s <%s>", o.gitName, o.gitEmail)
-	commitIfNeeded(fmt.Sprintf("determinize-ci-operator --current-release %s --future-release %s", o.currentRelease, strings.Join(o.futureReleases.Strings(), ",")), author)
-
-	cmd = "/usr/bin/config-brancher"
-	run(cmd, args...)
+	commitsCounter := 0
 
-	commitIfNeeded(fmt.Sprintf("config-brancher --current-release %s --future-release %s", o.currentRelease, strings.Join(o.futureReleases.Strings(), ",")), author)
-
-	cmd = "/usr/bin/ci-operator-prowgen"
-	args = []string{"--from-dir", "./ci-operator/config", "--to-dir", "./ci-operator/jobs"}
-	run(cmd, args...)
-
-	commitIfNeeded("ci-operator-prowgen --from-dir ./ci-operator/config --to-dir ./ci-operator/jobs", author)
-
-	cmd = "/usr/bin/ci-operator-config-mirror"
-	args = []string{"--config-path", "./ci-operator/config", "--to-org", "openshift-priv"}
-	run(cmd, args...)
-
-	commitIfNeeded("ci-operator-config-mirror --config-path ./ci-operator/config --to-org openshift-priv", author)
-
-	cmd = "/usr/bin/ci-operator-prowgen"
-	args = []string{"--from-dir", "./ci-operator/config", "--to-dir", "./ci-operator/jobs"}
-	run(cmd, args...)
-
-	commitIfNeeded("ci-operator-prowgen --from-dir ./ci-operator/config --to-dir ./ci-operator/jobs", author)
-
-	cmd = "/usr/bin/private-prow-configs-mirror"
-	args = []string{"--release-repo-path", "."}
-	run(cmd, args...)
-	commitIfNeeded("private-prow-configs-mirror --release-repo-path .", author)
-
-	cmd = "./hack/ensure_job_cluster.py"
-	args = []string{"-w=true"}
-	run(cmd, args...)
+	for i, step := range steps {
+		commited, err := runAndCommitIfNeeded(stdout, stderr, author, step.command, step.arguments)
+		if err != nil {
+			logrus.WithError(err).Fatalf("step[%d]: failed to run command and commit the changes", i)
```

Comment:
How exactly is the index in the log message useful here?

Reply:
well we have a `ci-operator-prowgen` step multiple times. I though that its nice to have information which step failed

Reply:
eh, not super useful to know index -- the command that failed will be in the error